examples/list_box_model/main.rs

```diff
@@ -0,0 +1,190 @@
+
+    application.connect_activate(build_ui);
+
+    application.run();
+}
+
+fn build_ui(application: &gtk::Application) {
+    let window = gtk::ApplicationWindow::builder()
+        .default_width(320)
+        .default_height(480)
+        .application(application)
+        .title("ListView: Custom Model")
+        .build();
+
+    let vbox = gtk::Box::new(gtk::Orientation::Vertical, 5);
+
+    // Create our list store and specify that the type stored in the
+    // list should be the RowData GObject we define at the bottom
+    let model = model::Model::new();
+
+    // And then create the UI part, the listbox and bind the list store
+    // model to it. Whenever the UI needs to show a new row, e.g. because
+    // it was notified that the model changed, it will call the callback
+    // with the corresponding item from the model and will ask for a new
+    // gtk::ListBoxRow that should be displayed.
+    //
+    // The gtk::ListBoxRow can contain any possible widgets.
+    let listbox = gtk::ListBox::new();
+    listbox.bind_model(Some(&model),
+        clone!(@weak window => @default-panic, move |item| {
+            let box_ = gtk::ListBoxRow::new();
+            let item = item.downcast_ref::<RowData>().expect("Row data is of wrong type");
+
+            let hbox = gtk::Box::new(gtk::Orientation::Horizontal, 5);
+
+            // Create the label and spin button that shows the two values
+            // of the item. We bind the properties for the two values to the
+            // corresponding properties of the widgets so that they are automatically
+            // updated whenever the item is changing. By specifying SYNC_CREATE the
+            // widget will automatically get the initial value of the item set.
+            //
+            // In case of the spin button the binding is bidirectional, that is any
+            // change of value in the spin button will be automatically reflected in
+            // the item.
+            let label = gtk::Label::new(None);
+            item.bind_property("name", &label, "label")
+                .flags(glib::BindingFlags::DEFAULT | glib::BindingFlags::SYNC_CREATE)
+                .build();
+            hbox.append(&label);
+
+            let spin_button = gtk::SpinButton::with_range(0.0, 100.0, 1.0);
+            item.bind_property("count", &spin_button, "value")
+                .flags(glib::BindingFlags::DEFAULT | glib::BindingFlags::SYNC_CREATE | glib::BindingFlags::BIDIRECTIONAL)
+                .build();
+            hbox.append(&spin_button);
+
+            // When the edit button is clicked, a new modal dialog is created for editing
+            // the corresponding row
+            let edit_button = gtk::Button::with_label("Edit");
+            edit_button.connect_clicked(clone!(@weak window, @strong item => move |_| {
+            let dialog = gtk::Dialog::with_buttons(Some("Edit Item"), Some(&window), gtk::DialogFlags::MODAL,
+                &[("Close", ResponseType::Close)]);
```

Comment:
Why doesn't `cargo fmt` complain about this? :)

```suggestion
            let dialog = gtk::Dialog::with_buttons(
                Some("Edit Item"),
                Some(&window),
                gtk::DialogFlags::MODAL,
                &[("Close", ResponseType::Close)],
            );
```

Reply:
The code is inside a proc macro(clone! is a proc macro right?) so I imagine rustfmt just gives up since it can't know if the tokens inside the proc macro are valid rust at all

Reply:
Oh no, yes you're right. I missed that this is inside `clone!`. (FWIW this is exactly why I don't use the macro in my own code)